Gemfile

```diff
@@ -28,22 +28,14 @@ group :system_tests do
   gem 'beaker-puppet_install_helper', :require => false
 end
 
-is_x64 = Gem::Platform.local.cpu == 'x64'
-if is_x64
-  platform(:x64_mingw) do
-    gem "win32-dir", "~> 0.4.9", :require => false
-    gem "win32-process", "~> 0.7.4", :require => false
-    gem "win32-service", "~> 0.8.4", :require => false
-    gem "minitar", "~> 0.5.4", :require => false
-  end
-else
-  platform(:mingw) do
-    gem "win32-process", "~> 0.6.5", :require => false
-    gem "win32-service", "~> 0.7.2", :require => false
-    gem "minitar", "~> 0.5.4", :require => false
-    if RUBY_VERSION =~ /^1\./ then
-      gem "win32console", :require => false
-    end
+platforms :mswin, :mingw do
```

Comment:
Now, going back to 3.3.0, because that is the earliest this module supports:

```
c:\code\puppetlabs-registry> SET GEM_PUPPET_VERSION=3.3.0
c:\code\puppetlabs-registry> ruby -v
ruby 1.9.3p545 (2014-02-24) [i386-mingw32]
c:\code\puppetlabs-registry> bundle install
c:\code\puppetlabs-registry> type Gemfile.lock
    puppet (3.3.0)
      facter (~> 1.6)
      hiera (~> 1.0)
      rgen (~> 0.6.5)
    win32-dir (0.5.0)
      ffi (>= 1.0.0)
    win32-process (0.8.0)
      ffi (>= 1.0.0)
    win32-security (0.2.5)
      ffi
    win32-service (0.8.6)
      ffi
    win32console (1.3.2-x86-mingw32)

PLATFORMS
  x86-mingw32
```

Note that the Gemfile is now doing more control of the dependencies.

Also note that the versions of the Gems are not technically correct. However to become more correct, we'd need to add the Gemfile dependencies in a nice complicated scenario based on each Puppet version below 3.5.x and I just would rather let them move up in this way and deal with errors if we come across any in specs (which haven't come up yet as valid reasons for failure in any place I've converted). Since we don't test against the oldest version supported, this section is really for any community members who may be.